com.civicrm.hrjobroles/CRM/Hrjobroles/BAO/HrJobRoles.php

```diff
@@ -58,28 +72,14 @@ public static function buildDbOptions($fieldName) {
     $query = "SELECT cpv.value, cpv.label from civicrm_option_value cpv
               LEFT JOIN civicrm_option_group cpg on cpv.option_group_id = cpg.id
               WHERE cpg.name = %1";
-    $options = array();
+    $options = [];
     $result = CRM_Core_DAO::executeQuery($query, $queryParam);
     while ($result->fetch()) {
-      $options[] =  array( 'value'=>$result->value, 'label'=>strtolower($result->label) );
+      $options[$result->value] = strtolower($result->label);
     }
```

Comment:
the output of buildOptions  is different from buildDbOptions  ... buildOptions get the result from the cache where buildDbOptions get it from the database ... so if you have added some option values via  sql query  then buildOptions will not be able to output the new inserted option value even if you cleared the cache , the only way to retrieve this newly inserted option value is to add it via this method CRM_Core_BAO_OptionValue::add so in this case it will be added to the cache and you can use buildOptions to retreive all the option values including it ... but unfortunately using CRM_Core_BAO_OptionValue::add in the tests to add option values was always cause civix to hit the time limit .